app/services/foreman_bootdisk/iso_generator.rb

```diff
@@ -163,6 +163,14 @@ def self.generate(opts = {})
       raise Foreman::Exception.new(N_('ISO hybrid conversion failed: %s'), $?) unless system(*["isohybrid", isohybrod_args, iso].flatten.compact)
 
       yield iso
+    ensure
+      # Clean the working directory (not the ISO file itself)
+      FileUtils.rm_f(File.join(wd, 'build'))
+      # Temporary directory cannot be cleaned in-process due to asynchronous send_file call.
+      # Also we cannot rely on systemd-tmpfiles-clean as private temporary files are not subject
+      # of scheduled cleanups. Let's clean bootdisks from prevous requests manually by finding
+      # and deleting all directories created 30 minutes ago.
+      Rails.root.glob('tmp/bootdisk-iso-*').select { |f| File.ctime(f) < (Time.now.to_i - (60 * 30)) }.each { |f| FileUtils.rm_f(f) }
```

Comment:
Maybe it should store the time in a variable rather than recalculating it inside the loop?
```suggestion
      delete_older_than = Time.now.to_i - (60 * 30)
      Rails.root.glob('tmp/bootdisk-iso-*').select { |f| File.ctime(f) < delete_older_than }.each { |f| FileUtils.rm_f(f) }
```